This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Mon, 30 Nov 2015 17:19:26 +0100
Subject: [PATCH 1/2] Fix: test arpa/inet.h presence before using it

Fix: test arpa/inet.h presence before using it

This fixes the Windows build.

Taken from:
https://git.xiph.org/?p=icecast-libshout.git;a=commit;h=53aa028d13ac624e2c1e71796d529e773867d1d4

diff --git a/configure.ac b/configure.ac
index 1111111..2222222 100644
--- a/configure.ac
+++ b/configure.ac
@@ -74,7 +74,7 @@ dnl Checks for programs.
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_HEADER_TIME
-AC_CHECK_HEADERS([strings.h sys/timeb.h])
+AC_CHECK_HEADERS([strings.h sys/timeb.h arpa/inet.h])
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
diff --git a/src/proto_roaraudio.c b/src/proto_roaraudio.c
index 1111111..2222222 100644
--- a/src/proto_roaraudio.c
+++ b/src/proto_roaraudio.c
@@ -28,7 +28,9 @@
 #endif
 
 /* for htonl(). */
+#ifdef HAVE_ARPA_INET_H
 #include <arpa/inet.h>
+#endif
 
 #include <stdio.h>
 #include <stdlib.h>

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tony Theodore <tonyt@logyst.com>
Date: Sun, 25 Feb 2018 14:53:34 +1100
Subject: [PATCH 2/2] openssl 1.1 fix


diff --git a/src/tls.c b/src/tls.c
index 1111111..2222222 100644
--- a/src/tls.c
+++ b/src/tls.c
@@ -65,7 +65,6 @@ static inline int tls_setup(shout_tls_t *tls)
 
 	SSL_library_init();
 	SSL_load_error_strings();
-	SSLeay_add_all_algorithms();
  	SSLeay_add_ssl_algorithms();
 
 	meth = TLSv1_client_method();
